src/CreeDictionary/API/search/runner.py

```diff
@@ -43,4 +48,7 @@ def search(
     if cvd_search_type.should_do_search():
         do_cvd_search(search_run)
 
+    if search_run.query.eip:
+        eip_search.inflect_search_results()
```

Comment:
[continued]

...THEN, phase 2 inflects the search results, but only after a whole bunch of other search stuff happens?

These two phases are discontiguous and rely on mutating the `replace_query()`, correct?

Reply:
That is correct. If you search for `they swam`, phase 1 needs to turn that into `swim` before generating any other search results, and then inflect the results after.

Now that I write that out, that almost reminds of what a `with` statement does, and I see that this could be made into a clearer wrapper function. In pseudocode:

    def do_eip_search(search_run, do_normal_search):
        if not query_analyses:
            do_normal_search(search_run)
            return
        
        uninflected_search_run = SearchRun(uninflected_query, …)
        do_normal_search(uninflected_search_run)
        for result in uninflected_search_run.results:
            if is_inflectable(result):
                search_run.add_result(inflect(result))
            else:
                search_run.add_result(result)

PS: no malice was involved in not (yet) addressing your comments on https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/pull/861

Reply:
Haha, no worries! I just wanted to make sure I understood how the two phase thing works here!